### PR TITLE
Support typescript when parsing GraphQL queries

### PR DESCRIPTION
**Pull request by dennari**
Also extract ast parsing into a utility function

### CODE REVIEW

File: packages/gatsby/src/bootstrap/resolve-module-exports.js [hunk 1]
```diff
@@ -24,25 +24,7 @@ module.exports = (modulePath, resolver = require.resolve) => {
   }
   const code = fs.readFileSync(absPath, `utf8`) // get file contents
 
-  const babylonOpts = {
```
Comment (dennari) [inline packages/gatsby/src/bootstrap/resolve-module-exports.js]: This is now using the larger set of plugins. Hopefully that's not a problem?

**Comment by dennari** [comment]
I'm going to be away for two weeks, so please feel free to modify this PR as needed and merge it in.

**Comment by deltaskelta** [comment]
its working for me

**Comment by m-allanson** [comment]
Published as `gatsby@2.0.0-beta.28`

**Pull request merged**